feat(leadspace): updating to v2

IgnacioBecerra opened the pull request:
### Related Ticket(s)
[#10788](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10788)

### Description
This PR adds the new `highlight` and `type-style` properties to the `c4d-leadspace-heading` component, as well as takes care of some of the spacing updates from the latest design specs.

### Changelog

**New**

- adds the new `highlight` and `type-style` properties for leadspace heading

**Changed**

- updated spacing and story knobs

**Removed**

- centered leadspace stories

<!-- React and Web Component deploy previews are enabled by default. -->
<!-- To enable additional available deploy previews, apply the following -->
<!-- labels for the corresponding package: -->
<!-- *** "test: e2e": Codesandbox examples and e2e integration tests -->
<!-- *** "package: services": Services -->
<!-- *** "package: utilities": Utilities -->
<!-- *** "RTL": React / Web Components (RTL) -->
<!-- *** "feature flag": React / Web Components (experimental) -->

ariellalgilmore commented on packages/styles/scss/components/leadspace/_leadspace.scss:
```diff
@@ -22,8 +20,56 @@
 $btn-min-width: 26;
 
+$leadspace-heading-highlight: (
+  fallback: #0F62FE,
+  values: (
+    (
+      theme: themes.$white,
+      value: #0F62FE,
+    ),
+    (
+      theme: themes.$g10,
+      value: #0F62FE,
+    ),
+    (
+      theme: themes.$g90,
+      value: #78A9FF,
+    ),
+    (
+      theme: themes.$g100,
+      value: #78A9FF,
```
```suggestion
  fallback: $blue-60,
  values: (
    (
      theme: themes.$white,
      value: $blue-60,
    ),
    (
      theme: themes.$g10,
      value: $blue-60,
    ),
    (
      theme: themes.$g90,
      value: $blue-40,
    ),
    (
      theme: themes.$g100,
      value:$blue-40,
```

ariellalgilmore commented on packages/styles/scss/components/leadspace/_leadspace.scss:
```diff
@@ -473,13 +563,48 @@ $btn-min-width: 26;
+  :host(#{$c4d-prefix}-leadspace)[size='super'] {
+    @include breakpoint(lg) {
+      .#{$prefix}--leadspace__overlay {
+        height: 640px;
```
should we convert this to rem?
```suggestion
        height: rem(640px);
```

ariellalgilmore replied:
oh yeah i have seen that! you might be missing the utility import

ariellalgilmore replied:
oh it should be this import `@use '@carbon/styles/scss/utilities' as *;`

ariellalgilmore commented on packages/styles/scss/components/leadspace/_leadspace.scss:
```diff
@@ -473,13 +563,48 @@ $btn-min-width: 26;
+  :host(#{$c4d-prefix}-leadspace)[size='tall'] {
+    @include breakpoint(lg) {
+      .#{$prefix}--leadspace__overlay {
+        height: 560px;
```
```suggestion
        height: rem(560px);
```

ariellalgilmore commented on packages/styles/scss/components/leadspace/_leadspace.scss:
```diff
@@ -473,13 +563,48 @@ $btn-min-width: 26;
+
+    @include breakpoint(lg) {
+      .#{$prefix}--leadspace__overlay {
+        height: 480px;
```
```suggestion
        height: rem(480px);
```

ariellalgilmore commented on packages/web-components/.storybook/container.scss:
```diff
@@ -15,10 +15,12 @@
 
 @use '@carbon/ibmdotcom-styles/scss/internal/callout/tokens' as callout-tokens;
 @use '@carbon/styles/scss/components/button/tokens' as button-tokens;
+@use '@carbon/ibmdotcom-styles/scss/components/leadspace' as leadspace;
```
wondering if we can seperate out the tokens like callout out does in separate file, so they only need to import those if they use a different theme. and then also add it to the dotcom [themes cdn file ](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/feat/carbon-for-ibm-dotcom-v2/packages/web-components/src/globals/scss/themes.scss)

ariellalgilmore commented on packages/web-components/src/components/leadspace/leadspace.scss:
```diff
@@ -8,73 +8,6 @@
 @use '@carbon/styles/scss/config' as *;
 @use '@carbon/styles/scss/spacing' as *;
-@use '@carbon/ibmdotcom-styles/scss/components/leadspace';
+@use '../../../../styles/scss/components/leadspace';
```
```suggestion
@use '@carbon/ibmdotcom-styles/scss/components/leadspace';
```

ariellalgilmore commented on packages/web-components/src/components/leadspace/leadspace.scss:
```diff
@@ -8,73 +8,6 @@
-    background-position: 0% 50%;
-  }
-}
+@use '@carbon/ibmdotcom-styles/scss/globals/vars' as *;
```
```suggestion

```

ariellalgilmore commented on packages/web-components/src/components/leadspace/leadspace.ts:
```diff
@@ -149,6 +149,10 @@ class C4DLeadSpace extends StableSelectorMixin(LitElement) {
     });
+
+    if(this.size === 'short') {
+      this.querySelector('c4d-leadspace-heading')?.setAttribute('type-style', 'fluid-heading-05');
```
```suggestion
      this.querySelector(`${c4dPrefix}-leadspace-heading`)?.setAttribute('type-style', 'fluid-heading-05');
```

ariellalgilmore reviewed (commented):
The highlight looks awesome!! just some cleanup comments

RichKummer commented:
Hey @IgnacioBecerra , the highlight and type swapping look great! I noticed a few spacing issues `@sm` and `@md`:

1. The spacing looks off at the smaller breakpoints. There should be `48px` between the heading and body copy. Finally, there should be `32px` spacing between the button group and the image/video if there is one. It looks like the spacing may be stacking when it renders?

<img width="324" alt="Screenshot 2023-09-21 at 5 35 42 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/1707bda8-d1e5-46f2-a1ac-573203b89e1c">

2. When `@sm`, the body copy runs off the view instead of wrapping at the edge. It looks like this area is scrollable horizontally, not sure what's happening at this breakpoint 🤔 
<img width="498" alt="Screenshot 2023-09-21 at 5 38 35 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/27df92e7-9c98-42b1-a5e4-8df242fa3bda">

3. The `button group` looks very condensed compared to [Olivia's button group ](https://www.figma.com/file/HFWHrSPZWdHd8ilqR5QFAk/Button-group?type=design&node-id=119%3A3358&mode=design&t=7BSxiyCMJRemo0uE-1)updates. There may be a separate PR for this but just want to make sure something in `lead space` isn't effecting it here.
<img width="324" alt="Screenshot 2023-09-21 at 5 35 42 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/e93a7c03-496a-4b58-86bd-1ccaf6e0e86a">

IgnacioBecerra commented:
@RichKummer Addressed comments!

ariellalgilmore reviewed (approved):
just a few small things, but otherwise LGTM!

RichKummer reviewed (commented):
@IgnacioBecerra , the vertical spacing and button group updates look great! I noticed a few things about horizontal spacing:

1. For `@sm`, the heading looks like it is being condensed too much. It should extend the full width of the container (ultimately with `16px` on both left and right for `@sm`.
<img width="443" alt="Screenshot 2023-09-26 at 3 23 32 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/d00c22b0-4995-4191-ad76-136bfdbcda6a">

Specs:
<img width="509" alt="Screenshot 2023-09-26 at 3 27 41 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/d654acfb-2338-4376-ad9e-bf3addd8def8">

2. When `@md`, both the heading and subheading should extend six columns - 32px. It looks like this is a little off (or at the very least they're not equal).
<img width="683" alt="Screenshot 2023-09-26 at 3 29 55 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/39b2a15e-bf2e-4be7-a812-7de215e3294c">

Specs:
<img width="1216" alt="Screenshot 2023-09-26 at 3 29 32 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/d1f9a863-b127-439f-8540-7542e9fd8690">

3. The height of `Super` is coming up as `672px` when it should be `640px` on desktop sizes. The others are coming in with a greater height than they should as well. Tall shoiuld be 560px, Medium should be 480px, and Short should be 320px. 
<img width="1077" alt="Screenshot 2023-09-26 at 3 33 09 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/0114f925-8ca2-438c-84f6-6a969cf04e69">

4. The subheading should be more condensed width wise at desktop sizes. It is rendering as `#text` which might be off when compared to the heading? See [design specs:](https://www.figma.com/file/Gfb6xdHwKdBgNloNqlSbnG/Lead-space-v2?type=design&node-id=1015%3A13228&mode=design&t=n7frr2iNacUrw1Yk-1)

<img width="1123" alt="Screenshot 2023-09-26 at 3 36 42 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/15643582/bcaf043f-fc7c-4d05-9f59-22b6ead68a14">

IgnacioBecerra commented:
@RichKummer addressed your comments! I might've been checking the inner container for the heights instead of the whole component since it was adding an extra 32px from the top padding haha.

RichKummer commented:
@IgnacioBecerra this is looking great! The vertical spacing issues look resolved. I did notice that the `Short` size for lead space is coming in at `480px` height when it should render with `320px` height.

RichKummer reviewed (approved):
Thanks for the last minute fix @IgnacioBecerra , this one LGTM 🔥 🔥 🔥

kennylam commented on packages/styles/scss/components/leadspace/_tokens.scss:
```diff
@@ -0,0 +1,44 @@
+
+$btn-min-width: 26;
+
+$leadspace-heading-highlight: (
```
I think all of the highlight color token logic could be removed by simply using `$link-primary`, no? See the [Link](https://carbondesignsystem.com/guidelines/color/tokens) section of the color guidance page.

IgnacioBecerra replied:
That's true! I didn't realize there was already a token for that color mapping. I'll remove the token file and use this token instead

fyi @RichKummer, the [specs](https://www.figma.com/file/Gfb6xdHwKdBgNloNqlSbnG/Lead-space-v2?type=design&node-id=1021-36034&mode=design&t=lGZeOZtq02WhoAAA-4) can be updated to include the token instead 😄 👍

kennylam reviewed (commented):
Looks great @IgnacioBecerra! Just a few comments.

kennylam reviewed (approved):
LGTM!
